lib/ReplacePointerBitcastPass.cpp

```diff
@@ -474,6 +474,14 @@ clspv::ReplacePointerBitcastPass::run(Module &M, ModuleAnalysisManager &) {
         }
 
         if (source_ty && dest_ty && source_ty != dest_ty) {
+          // LLVM can elide the GEP of the first element of a vector when using
+          // opaque pointers, skip this case
+          if (auto *source_vec_ty = dyn_cast<VectorType>(source_ty)) {
```

Comment:
There seems to be a mismatch between this change and the one in UndoInstCombine that only looks for global variables.